cmd/auth/logout.go

```diff
@@ -3,26 +3,131 @@
 package auth
 
 import (
+	"fmt"
+
 	"github.com/pingidentity/pingcli/cmd/common"
+	auth_internal "github.com/pingidentity/pingcli/internal/auth"
+	"github.com/pingidentity/pingcli/internal/commands/auth"
+	"github.com/pingidentity/pingcli/internal/configuration/options"
+	"github.com/pingidentity/pingcli/internal/customtypes"
+	"github.com/pingidentity/pingcli/internal/profiles"
 	"github.com/spf13/cobra"
 )
 
+// NewLogoutCommand creates a new logout command that clears stored credentials
 func NewLogoutCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Args:                  common.ExactArgs(0),
 		DisableFlagsInUseLine: true, // We write our own flags in @Use attribute
-		Long:                  "Logout user from the CLI",
+		Long:                  "Logout user from the CLI by clearing stored credentials. Credentials are cleared from both keychain and file storage. By default, uses the authentication method configured in the active profile. You can specify a different authentication method using the auth method flags.",
 		RunE:                  authLogoutRunE,
 		Short:                 "Logout user from the CLI",
 		Use:                   "logout [flags]",
 	}
 
+	// Add the same auth method flags as login command
+	cmd.Flags().AddFlag(options.AuthMethodAuthCodeOption.Flag)
+	cmd.Flags().AddFlag(options.AuthMethodClientCredentialsOption.Flag)
+	cmd.Flags().AddFlag(options.AuthMethodDeviceCodeOption.Flag)
+
+	// These flags are mutually exclusive - only one can be specified
+	cmd.MarkFlagsMutuallyExclusive(
+		options.AuthMethodAuthCodeOption.Flag.Name,
+		options.AuthMethodClientCredentialsOption.Flag.Name,
+		options.AuthMethodDeviceCodeOption.Flag.Name,
+	)
+
 	return cmd
 }
 
+// authLogoutRunE implements the logout command logic, clearing credentials from both
+// keychain and file storage for the specified or configured authentication method
 func authLogoutRunE(cmd *cobra.Command, args []string) error {
```

Comment:
Generally same comments from login apply here

- Flag parsing logic
- Moving command logic to internal package 
- Returning PingCLIErrors instead of fmt.errorf